### PR TITLE
MNT: Move pooch to a test dependency

### LINKED ISSUE #247: `pooch` runtime dependency intended?
### Comment:

Hi. PR #221 made `pooch` an install dependency of `scipy`.

https://github.com/conda-forge/scipy-feedstock/blob/3a08a185e651cecdc19023d04c22677b5bdc5d26/recipe/meta.yaml#L105-L110

Was this intended? I assume no, as `pooch` is not a [formal requirement of `scipy`](https://github.com/scipy/scipy/blob/986f248024afb0dc3646d43b963dfe43d9e3cdbb/pyproject.toml#L47-L51), but maybe it was as it was added in the very first commit in PR #221.

From a downstream users view (c.f. https://github.com/conda-forge/pyhf-feedstock/issues/21) it was just very strange to see an unexpected dependency getting installed with no runtime requirement link.

```console
$ docker run --rm -ti mambaorg/micromamba:1.4.9-bullseye-slim
(base) mambauser@29855946c848:/tmp$ micromamba install --yes --channel conda-forge scipy
(base) mambauser@29855946c848:/tmp$ python -m pip show scipy | grep Req
Requires: numpy
Required-by: 
(base) mambauser@29855946c848:/tmp$ python -m pip show pooch | grep Req
Requires: packaging, platformdirs, requests
Required-by: 
(base) mambauser@29855946c848:/tmp$ 
```

cc @h-vetinari

### PR DESCRIPTION
**Pull request by matthewfeickert**
* Resolves #247
* Resolves https://github.com/conda-forge/pyhf-feedstock/issues/21

Remove `pooch` as an install dependency to avoid distributing optional dependencies without user request.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [N/A] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -233,6 +231,8 @@ outputs:
         - pytest-timeout
         - pytest-xdist    # [not (win and python_impl == "pypy")]
         - mpmath
+        # needed for scipy.datasets
+        - pooch
```
Comment (matthewfeickert) [inline recipe/meta.yaml:235]: Moving to `test` `requires` as requested in https://github.com/conda-forge/scipy-feedstock/issues/247#issuecomment-1710916764.

**Comment by matthewfeickert** [comment]
Thank you for already taking actions to mitigate the failures, @h-vetinari. :pray:

**Comment by h-vetinari** [comment]
OK, this is currently some pypy variants, so let's put this in. Thanks for the issue & pr @matthewfeickert!

**Comment by matthewfeickert** [comment]
Thanks very much for fixing everything @h-vetinari, as well as for your help, information, and direction on all of this.

**Pull request merged**